core/templates/pages/contributor-dashboard-page/opportunities-list/opportunities-list.component.ts

```diff
@@ -45,8 +45,10 @@ export class OpportunitiesListComponent {
   @Input() opportunityHeadingTruncationLength: number;
   @Input() opportunityType: string;
 
+  private static readonly DEFAULT_LAST_PAGE_NUMBER = 1000;
```

Comment:
Maybe add a comment to explain what this means, it's not really obvious TBH.

I also wonder, why do we need to show a default last page number? Would it be better to show the correct one from the outset, and not show the page numbers before things load (like, can we put stuff behind a loading message)? Otherwise it seems kinda misleading, especially since the 1000 has no real connection to reality.

Also does this mean that the "last" page number continues increasing over time as more opps are fetched?

Reply:
Sure, added a comment.

We don't surface the last page number to the user, rather it's mostly an implementation detail for how we determine whether to disable the "next page" button. The button is disabled if `loadingOpportunityData || activePageNumber >= lastPageNumber`.

The "last" page number is updated as follows:
1. Set to 1000 on first initialization.
2. After each opportunity batch fetch: `this.lastPageNumber = more ? this.lastPageNumber : Math.ceil(
          this.opportunities.length / this.OPPORTUNITIES_PAGE_SIZE);`